File: Programming and Projection Methods/Lab8C#/Laborator8/CompanyServices/IServer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CompanyModel;

namespace CompanyServices
{
    public interface IServer
    {
        void Login(User u, IObserver o);
        void Logout(User u, IObserver o);
        RBooking[] FindAllRBookings(String dest, String date, String hour);
        String BookPlaces(Ride ride, Client client, int nrplaces);
        Ride[] FindAllRides();
        void NotifyObservers();
    }
}

File: Programming and Projection Methods/Lab7/Laborator7/CompanyServices/src/main/java/company/services/IServer.java
package company.services;

import company.model.Client;
import company.model.RBooking;
import company.model.Ride;
import company.model.User;

public interface IServer {
    void login(User u, IObserver o);

    void logout(User u, IObserver o);

    RBooking[] findAllRBookings(String dest, String date, String hour);

    String bookPlaces(Ride ride, Client client, int nrplaces);

    Ride[] findAllRides();

    void notifyObservers();
}
